src/OpenTelemetry.Exporter.OpenTelemetryProtocol.PersistentStorage/OtlpTraceExporterPersistentStorageExtensions.cs

```diff
@@ -0,0 +1,77 @@
+// </copyright>
+
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Extensions.PersistentStorage.Abstractions;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace;
+
+/// <summary>
+/// Extension methods to simplify registering of the OpenTelemetry Protocol (OTLP) exporter
+/// with persistent storage.
+/// </summary>
+public static class OtlpTraceExporterPersistentStorageExtensions
+{
+    /// <summary>
+    /// Adds OpenTelemetry Protocol (OTLP) exporter to the TracerProvider
+    /// with access to persistent storage.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> builder to use.</param>
+    /// <param name="name">Name which is used when retrieving options.</param>
+    /// <param name="configure">Callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+    /// <param name="persistentStorageFactory">Factory function to create a <see cref="PersistentBlobProvider"/>.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddOtlpExporterWithPersistentStorage(
+        this TracerProviderBuilder builder,
+        string name,
+        Action<OtlpExporterOptions> configure,
+        Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
+    {
+        Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNull(persistentStorageFactory);
+
+        builder.ConfigureServices(services =>
+        {
+            // Try this if it doesn't work there are other ways to plug into options
+            services
+                  .AddOptions<OtlpExporterOptions>()
+                  .Configure<IServiceProvider>((otlpExporterOptions, serviceProvider) =>
+                  {
+                      otlpExporterOptions.PersistentBlobProvider = persistentStorageFactory(serviceProvider);
+                  });
+        });
+
+        return builder.AddOtlpExporter(name, configure);
+    }
+
+    /// <summary>
+    /// Adds OpenTelemetry Protocol (OTLP) exporter to the TracerProvider
+    /// with access to persistent storage.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> builder to use.</param>
+    /// <param name="configure">Callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+    /// <param name="persistentStorageFactory">Factory function to create a <see cref="PersistentBlobProvider"/>.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddOtlpExporterWithPersistentStorage(
+        this TracerProviderBuilder builder,
+        Action<OtlpExporterOptions> configure,
+        Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
+    {
+        return builder.AddOtlpExporterWithPersistentStorage(configure, persistentStorageFactory);
```

Comment:
This is calling itself in an infinite loop 😄 You probably meant to do...

```csharp
return builder.AddOtlpExporterWithPersistentStorage(name: null, configure, persistentStorageFactory);
```

Reply:
🤫 oh no...